tests/test_cmd_list_depends.py

```diff
@@ -70,6 +75,37 @@ def test_loop():
     )
 
 
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="Requires python3.9 or higher")
+def test_loop_topological():
+    addons_set = mock_addons_set(
+        {
+            "a": {"depends": ["b"]},
+            "b": {"depends": ["c"]},
+            "c": {"depends": ["a"]},
+        }
+    )
+    addons_selection = mock_addons_selection("a")
+    try:
+        list_depends_command(
+            addons_selection,
+            addons_set,
+            transitive=True,
+            addon_sorter=AddonSorterTopological(),
+        )
+    except click.exceptions.Exit as e:
+        assert e
```

Comment:
You can use `with pytest.raise`.  Can you also subclass `typer.Exit` as `CycleErrorExit` or something to make the test more precise?